From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Shelley Vohr <shelley.vohr@gmail.com>
Date: Thu, 13 Sep 2018 09:08:10 -0700
Subject: FIXME: remove async_id assertion check

async hooks are hella broken in Electron. This was checking that they weren't,
but they are, so we just disabled the check. YOLO.

diff --git a/src/api/callback.cc b/src/api/callback.cc
index c8934e1cd33a364ba55b437b080fe2c1c0f5de6c..63c9f5d7c08bebd25bd2602ab6b97c8987a102a4 100644
--- a/src/api/callback.cc
+++ b/src/api/callback.cc
@@ -117,12 +117,14 @@ void InternalCallbackScope::Close() {
     perform_stopping_check();
   }
 
+#if 0  // FIXME(codebytere): figure out why this check fails/causes crash
   // Make sure the stack unwound properly. If there are nested MakeCallback's
   // then it should return early and not reach this code.
   if (env_->async_hooks()->fields()[AsyncHooks::kTotals]) {
     CHECK_EQ(env_->execution_async_id(), 0);
     CHECK_EQ(env_->trigger_async_id(), 0);
   }
+#endif
 
   if (!tick_info->has_tick_scheduled() && !tick_info->has_rejection_to_warn()) {
     return;
